### ISSUE TITLE
Errors imported from .Retries queues have required FailedQ header stripped when retrying

### ISSUE THREAD

**Issue by boblangley** [Particular/ServiceControl#913]
The secondary retries processors require a `FailedQ` header that points to the normal input queue for the endpoint. If a `FailedQ` header is not present the message is rejected and forwarded to the system error queue.

Additionally, when a message fails in the secondary retries processor, it is forwarded to the system error queue with the `.Retries` address set to the `FailedQ` header value.

Service Control strips the `FailedQ` header when sending a message back for retry. 

The result is, for a message that originally fails in the `.Retries` queue, and is then retried from Service Control, the message will never able to be retried being immediately returned to the system error queue. The message is now stuck in a loop:

![image](https://cloud.githubusercontent.com/assets/2546913/22756994/f94d8a8c-edfd-11e6-94dc-9e914aae6658.png)

**Comment by dbelcham** [comment]
This is a bug that we currently have an open support case for. As it stands the bug prevents the client from being able to successfully retry their messages.

**Comment by SzymonPobiega** [comment]
Given my comment [here](https://github.com/Particular/ServiceControl/pull/914#issuecomment-279382172), should we consider solving this on the endpoint level rather than SC level? It seems that it is a bug in NServiceBus recoverability.

**Comment by dbelcham** [comment]
Can it be solved in the endpoint? This is a bug in v5 (probably can be patched) and v4 (which we don't patch right?). There is no bug in v6.

**Comment by SzymonPobiega** [comment]
I am pretty sure this could be addressed via some header manipulation on the endpoint level or by using `IAdvancedSatellite`.

Another thing: this seems to be a a message loss scenario because user's won't be able to process the message.

**Comment by SzymonPobiega** [comment]
Here's my attempt to fix it on NSB level: https://github.com/Particular/NServiceBus/pull/4527

TL;DR it copies the `FailedQ` to a temporary header before sending to the `Retries` satellite and the `Retries` satellite uses either `FailedQ` or the temp header.

**Comment by SzymonPobiega** [comment]
@boblangley can you take a look at the PR above ☝️ . I think it solves the problem closer to where the root cause was.

**Comment by boblangley** [comment]
@SzymonPobiega The reason we were looking at fixing this in Service Control instead of NServiceBus is because we would need this fix for older versions of the Core as well. I believe the customer that raised the issue originally was on v4. If we won't back port it we will still need a fix on the Service Control side.

**Comment by boblangley** [comment]
Unless we just want to say for pre V5 you have to use Redirects. What do the other @Particular/servicecontrol-maintainers think?

**Comment by mikeminutillo** [comment]
Isn't part of the problem that once we've lost the original exception the data at rest (inside SC) is corrupt?

**Comment by boblangley** [comment]
@mikeminutillo Not that I am aware of, only that the original exception info is lost and replaced by the retries exception info. That is just how the headers are manipulated by the core, SC doesn't care as far as I know.

**Comment by SzymonPobiega** [comment]
So it seems the issue affects only NServiceBus 5 and earlier. The fix I proposed for V5 has been rejected by NServiceBus maintainers that said that a workaround (redirects) in ServiceControl is enough. Given that 4 years have passed and we haven't observed any instances of a problem, I propose to close this one.
